src/specmap/lib/index.js

```diff
@@ -45,15 +45,30 @@ function applyPatch(obj, patch, opts) {
     const valPatch = _get(patch.path)
     jsonPatch.apply(obj, [valPatch])
     const origValPatchValue = Object.assign({}, valPatch.value)
-    deepExtend(valPatch.value, patch.value)
-
-    // deepExtend doesn't merge arrays, so we will do it manually
-    for (const prop in patch.value) {
-      if (Object.prototype.hasOwnProperty.call(patch.value, prop)) {
-        const propVal = patch.value[prop]
-        if (Array.isArray(propVal)) {
-          const existing = origValPatchValue[prop] || []
-          valPatch.value[prop] = existing.concat(propVal)
+
+    const currentProps = Object.keys(valPatch.value.properties || {})
+    const newProps = Object.keys(patch.value.properties || {})
+
+    const hasConflictingProps = currentProps
```

Comment:
Would it be worthwhile to deepExtend the non-conflicting props, and assign the missing/conflicting ones?
This would replace the `if(hasConflictingProps)` and the `deepExtend` call.
```js
// if(hasFreelyNamedFields(patch.path) { ?? Something that looks at the path array to decide if its a freely named field.
// Do NOT deeply clone all object values, only those that need to be deeply merged.
for( const key in newProps ) {
  if(currentProps.hasOwnProperty(prop)) {
    deepExtend(currentProps[key], newProps[key]) 
  } else {
    Object.assign(currentProps[key], newProps[key])
  }
}
```

Reply:
Not sure I really follow the changes here, but I do think the logic could be simplified here a little. I'll make a few changes to simplify. 😄